### PR TITLE
[WIP] Adding environment variable for amplifier public address

### PR DESCRIPTION
**Pull request by bquenin**
ref #1018 

### How to test

    $ go install github.com/appcelerator/amp/cmd/amplifier
    $ export AMP_PUBLICHOST="this.is.a.test"
    $ amplifier
    amplifier (server version: , build: )
    2017/04/15 20:32:44 Configuration file successfully loaded
    Version: 
    Build: 
    Port: :50101
    PublicHost: this.is.a.test
    EtcdEndpoints: [http://etcd:2379]
    ElasticsearchURL: http://elasticsearch:9200
    NatsURL: nats://nats:4222
    DockerURL: unix:///var/run/docker.sock
    DockerVersion: 1.27
    2017/04/15 20:32:44 Connecting to Docker API at unix:///var/run/docker.sock version API: 1.27
    2017/04/15 20:32:44 Connected to Docker API at unix:///var/run/docker.sock
    2017/04/15 20:32:44 Listening on port: 50101

### CODE REVIEW

File: cmd/amplifier/server/configuration.go [hunk 1]
```diff
@@ -10,17 +10,17 @@ import (
 )
 
 const (
-	DefaultPort          = ":50101"
-	DefaultPublicAddress = "local.appcelerator.io"
-	DefaultTimeout       = time.Minute
+	DefaultPort       = ":50101"
+	DefaultPublicHost = "127.0.0.1"
```
Comment (ndegory) [inline cmd/amplifier/server/configuration.go:14]: we may have an issue once we're ready to do TLS termination outside of amplifier, if we use a different port, the exposed address won't match

Comment (bquenin) [inline cmd/amplifier/server/configuration.go]: This is the "inside the container" port, which is mapped anyway.

Comment (bquenin) [reply cmd/amplifier/server/configuration.go:14]: I can add an environment variable for port as well, or group both address and port into a single value, such as `PublicEndpoint`. Tell me what you think is best.

Comment (ndegory) [reply cmd/amplifier/server/configuration.go:14]: the way it's done most of the time is a URL, when the port is not explicitly set the CLI knows which port to try. We can image that the CLI first try TLS/443, and falls back to non TLS/50101. Unless the scheme and port are explicitly set.

Comment (bquenin) [reply cmd/amplifier/server/configuration.go:14]: gRPC is not really a well known scheme, and port guessing can be confusing for users. I'd rather set a global endpoint as "address:port" for clarity.

Comment (ndegory) [reply cmd/amplifier/server/configuration.go:14]: do we have access to the scheme and authority from the grpc metadata?

**Comment by bquenin** [comment]
Working on another solution for this issue

**Pull request closed without merging**